vesta/services/clash_of_code_entities.py

```diff
@@ -0,0 +1,213 @@
+import datetime
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import discord
+from discord import Embed
+
+
+class GameMode(Enum):
+    """
+    Represents a Clash of Code game mode
+    """
+
+    FASTEST = 0
+    REVERSE = 1
+    SHORTEST = 2
+
+    def __repr__(self):
+        return f"coc_mode_{self.name.lower()}"
+
+class Role(Enum):
+    """
+    Represents a Clash of Code player role
+    """
+
+    OWNER = 0
+    STANDARD = 1
+
+class State(Enum):
+    """
+    Represents a Clash of Code game state
+    """
+
+    PENDING = 0
+    RUNNING = 1
+    FINISHED = 2
+
+    def __repr__(self):
+        """
+        Gives the translation key for the state
+        """
+        return f"coc_game_state_{self.name.lower()}"
+
+    def __str__(self):
+        """
+        Gives the emoji for the state
+        """
+        return {
+            State.PENDING: "🕒",
```

Comment:
you really like emojis, don't you?

Reply:
I like them but here they serve the purpose of saying whether the player has finished the came or not

Reply:
(+ it adds legibility not to have only text)